File: services/ones-api/src/main/java/com/ones/api/domain/subscriptions/PaymentProfile.java
package com.ones.api.domain.subscriptions;

import java.time.Instant;
import java.util.Objects;

public class PaymentProfile {

    private final String userId;
    private final String mercadoPagoEmail;
    private final String country;
    private final String documentType;
    private final String documentNumber;
    private final String phoneNumber;
    private final String fullName;
    private final Instant createdAt;
    private final Instant updatedAt;
    private final Instant verifiedAt;

    public PaymentProfile(
            String userId,
            String mercadoPagoEmail,
            String country,
            String documentType,
            String documentNumber,
            String phoneNumber,
            String fullName,
            Instant createdAt,
            Instant updatedAt,
            Instant verifiedAt
    ) {
        this.userId = Objects.requireNonNull(userId);
        this.mercadoPagoEmail = mercadoPagoEmail;
        this.country = country;
        this.documentType = documentType;
        this.documentNumber = documentNumber;
        this.phoneNumber = phoneNumber;
        this.fullName = fullName;
        this.createdAt = Objects.requireNonNull(createdAt);
        this.updatedAt = Objects.requireNonNull(updatedAt);
        this.verifiedAt = verifiedAt;
    }

    public String getUserId() { return userId; }
    public String getMercadoPagoEmail() { return mercadoPagoEmail; }
    public String getCountry() { return country; }
    public String getDocumentType() { return documentType; }
    public String getDocumentNumber() { return documentNumber; }
    public String getPhoneNumber() { return phoneNumber; }
    public String getFullName() { return fullName; }
    public Instant getCreatedAt() { return createdAt; }
    public Instant getUpdatedAt() { return updatedAt; }
    public Instant getVerifiedAt() { return verifiedAt; }

    public PaymentProfile withUpdated(
            String mercadoPagoEmail,
            String country,
            String documentType,
            String documentNumber,
            String phoneNumber,
            String fullName,
            Instant updatedAt
    ) {
        return new PaymentProfile(
                userId,
                mercadoPagoEmail,
                country,
                documentType,
                documentNumber,
                phoneNumber,
                fullName,
                createdAt,
                updatedAt,
                verifiedAt
        );
    }
}

File: apps/ones_app/lib/features/subscriptions/domain/payment_profile.dart
class PaymentProfile {
  final String? userId;
  final String? mercadoPagoEmail;
  final String? country;
  final String? documentType;
  final String? documentNumber;
  final String? phoneNumber;
  final String? fullName;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  PaymentProfile({
    this.userId,
    this.mercadoPagoEmail,
    this.country,
    this.documentType,
    this.documentNumber,
    this.phoneNumber,
    this.fullName,
    this.createdAt,
    this.updatedAt,
  });

  PaymentProfile copyWith({
    String? mercadoPagoEmail,
    String? country,
    String? documentType,
    String? documentNumber,
    String? phoneNumber,
    String? fullName,
  }) {
    return PaymentProfile(
      userId: userId,
      mercadoPagoEmail: mercadoPagoEmail ?? this.mercadoPagoEmail,
      country: country ?? this.country,
      documentType: documentType ?? this.documentType,
      documentNumber: documentNumber ?? this.documentNumber,
      phoneNumber: phoneNumber ?? this.phoneNumber,
      fullName: fullName ?? this.fullName,
      createdAt: createdAt,
      updatedAt: updatedAt,
    );
  }

  factory PaymentProfile.fromJson(Map<String, dynamic> json) {
    return PaymentProfile(
      userId: json['userId'] as String?,
      mercadoPagoEmail: json['mercadoPagoEmail'] as String?,
      country: json['country'] as String?,
      documentType: json['documentType'] as String?,
      documentNumber: json['documentNumber'] as String?,
      phoneNumber: json['phoneNumber'] as String?,
      fullName: json['fullName'] as String?,
      createdAt: _parseDate(json['createdAt']),
      updatedAt: _parseDate(json['updatedAt']),
    );
  }

  Map<String, dynamic> toJson() => {
        if (mercadoPagoEmail != null) 'mercadoPagoEmail': mercadoPagoEmail,
        if (country != null) 'country': country,
        if (documentType != null) 'documentType': documentType,
        if (documentNumber != null) 'documentNumber': documentNumber,
        if (phoneNumber != null) 'phoneNumber': phoneNumber,
        if (fullName != null) 'fullName': fullName,
      };

  static DateTime? _parseDate(Object? s) {
    if (s is String && s.isNotEmpty) {
      return DateTime.tryParse(s)?.toLocal();
    }
    return null;
  }
}
